xcell/mappers/mapper_ACT_compsept.py

```diff
@@ -0,0 +1,65 @@
+from .mapper_ACT_base import MapperACTBase
+from pixell import enmap, reproject
+import numpy as np
+from scipy.interpolate import interp1d
+
+
+class MapperACTCompSept(MapperACTBase):
+    def __init__(self, config):
+        """
+        config - dict
+        {'file_map':'tilec_single_tile_D56_cmb_map_v1.2.0_joint.fits',
+         'file_mask':'tilec_mask.fits',
+         'file_noise': 'tilec_single_tile_D56_cmb_map_v1.2.0_joint_noise.fits',
+         'beam_file': 'tilec_single_tile_D56_cmb_map_v1.2.0_joint_beam.txt',
+         'mask_name': 'mask_ACTtsz',
+         'nside': 1024,
+         'lmax': 6000}
+        """
+        self._get_ACT_CompSept_defaults(config)
+
+    def _get_ACT_CompSept_defaults(self, config):
+        self._get_defaults(config)
+        self._get_ACT_defaults(config)
+        self.nl_coupled = None
+        self.beam_info = config.get('beam_info',
+                                    [{'type': 'custom',
+                                      'file': None}])
+
+    def _get_signal_map(self):
+        if self.signal_map is None:
+            # The 'Weights' FITS file contains the 2D Fourier space
+            # weight for each pixel corresponding to the detector array.
+            signal_map = enmap.read_map(self.file_map)
+            signal_map = reproject.healpix_from_enmap(signal_map,
+                                                      lmax=self.lmax,
+                                                      nside=self.nside)
+        return signal_map
+
+    def _get_mask(self):
+        if self.mask is None:
+            # It [The mask] has already been applied to the map.
+            # If you are doing a real-space analysis, you should
+            # exclude any pixels where the value in the mask is
+            # appreciably different from 1 since the signal there
+            # should be attenuated by the value in the mask.
+
+            self.pixell_mask = self._get_pixell_mask()
+            msk = reproject.healpix_from_enmap(self.pixell_mask,
+                                               lmax=self.lmax,
+                                               nside=self.nside)
+            msk[msk < 0.99] = 0
+        return msk
```

Comment:
I think this function is the same in `mapper_ACT_compsept.py` and in `mapper_ACTk.py`, so you could just put it in `mapper_ACT_base.py`.

Reply:
Not quite. For the compsept case we need line 51 while for the ACTk we don't.